### PR TITLE
Setup passport

### PR DESCRIPTION
**Pull request by Midlu**
Still adding passport

close #30

### CODE REVIEW

File: backend/routes/users.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+const router = express.Router();
+const knex = require('../db/knex');
+
+/* GET home page. */
+router.get('/createUser', (req, res, next) => {
```
Comment (signalnerve) [inline backend/routes/users.js]: GET /createUser seems weird - if this is nested at /users, you can just do `POST /`

File: backend/services/user.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const passport = require('passport');
+const LocalStrategy = require('passport-local');
+apassport.use(
```
Comment (signalnerve) [inline backend/services/user.js]: passport

File: backend/services/user.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      }
+      if (!user.verifyPassword(password)) {
+        return done(null, false);
+      }
+      return done(null, user);
+    });
+  })
+);
+
+module.exports = authen;
```
Comment (signalnerve) [inline backend/services/user.js]: missing function maybe?

**Comment by Midlu** [comment]
I barely know what I'm doing... hahahhahahahahahhahahahahhahahahahhahhaha...

**Comment by signalnerve** [comment]
@Midlu is this ready for review?

**Comment by Midlu** [comment]
Passport is somewhat implement, we still need to make a jwt to send back and secure routes, we can create users though

**Comment by Midlu** [comment]
Also, I pulled to get the entire repository up to date, I have changes from frontend

File: backend/services/User.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  const newUser = {
+    username,
+    email,
+    password,
+    is_admin
+  };
+  console.log('new user is ', newUser);
+  auth
+    .hashPassword(password)
+    .then(hashedPassowrd => {
```
Comment (signalnerve) [inline backend/services/User.js]: hashedPassword* but not a big deal

File: backend/services/User.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+const auth = require('./Auth');
+
+const createUser = (req, res, next) => {
+  let { password } = req.body;
```
Comment (signalnerve) [inline backend/services/User.js]: I think this could be in the const line below, right?

Comment (Midlu) [reply backend/services/User.js]: Yes, you are right i forgot that I put the password into a object like two lines below that

**Pull request merged**